src/shogun/base/SGObject.cpp

```diff
@@ -1092,26 +1092,23 @@ void CSGObject::init_auto_params()
 	}
 }
 
+CSGObject* CSGObject::get(const std::string& name, index_t index) const
+{
+	return sgo_details::get_by_tag(this, name, std::move(sgo_details::GetByNameIndex(index)));
+}
+
 CSGObject* CSGObject::get(const std::string& name, std::nothrow_t) const
     noexcept
 {
-	if (auto* result = get_sgobject_type_dispatcher<CDistance>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CKernel>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CFeatures>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CLabels>(name))
-		return result;
-	return nullptr;
+	return sgo_details::get_by_tag(this, name, std::move(sgo_details::GetByName()));
 }
 
 CSGObject* CSGObject::get(const std::string& name) const noexcept(false)
```

Comment:
oh and btw do you still need the non-exception version of this for the openml? I remember I asked you but I forgot what you said?

Reply:
Yes, I think that is still needed because there are issues catching errors in swig code

Reply:
ok cool Ill keep it then